### PR TITLE
Deprecate `autorollback` and `savepoint`when appending to reflect the API change

### LINKED ISSUE #105: Remove the “autorollback” arg from `appendDataset`
The feature has been deprecated in the Crunch API (appends now always automatically roll back in case of failure).

### PR DESCRIPTION
**Pull request by jonkeane**
Resolves #105

### CODE REVIEW

File: R/append-dataset.R [hunk 1]
```diff
@@ -5,14 +5,16 @@
 #' `dataset2` is not a Crunch dataset, it will be uploaded as a new
 #' dataset before appending. If it is a CrunchDataset, it may be subsetted with
 #' a filter expression on the rows and a selection of variables on the columns.
-#' @param autorollback logical: If the append fails, revert the dataset back
+#' @param autorollback Deprecated. This option no longer does anything.
+#' logical: If the append fails, revert the dataset back
```
Comment (nealrichardson) [inline R/append-dataset.R]: Remove this second line since it's no longer true

File: R/append-dataset.R [hunk 2]
```diff
@@ -5,14 +5,16 @@
...
 #' to its state before attempting to append? Default is \code{TRUE}, and you
 #' probably won't want to change that.
 #' @return `dataset1`, updated with `dataset2`, potentially filtered on rows and
 #' variables, appended to it.
 #' @export
-appendDataset <- function (dataset1, dataset2, autorollback=TRUE) {
+appendDataset <- function (dataset1, dataset2, autorollback) {
     stopifnot(is.dataset(dataset1))
+
```
Comment (nealrichardson) [inline R/append-dataset.R]:
```
if (!missing(autorollback)) {
    warning("The 'autorollback' argument is deprecated and has no effect", call.=FALSE)
}
```

and test for that warning.

File: R/batches.R [hunk 3]
```diff
@@ -1,22 +1,19 @@
-addBatch <- function (ds, ..., savepoint=TRUE, autorollback=savepoint, strict=TRUE, body=list(...)) {
+addBatch <- function (ds, ..., savepoint, autorollback, strict=TRUE, body=list(...)) {
```
Comment (nealrichardson) [inline R/batches.R]: This function is not exported, so remove the arguments here, and handle the removal/deprecation warning in the calling function(s).

File: R/batches.R [hunk 4]
```diff
@@ -1,22 +1,19 @@
...
-        ## Just execute and let the "Result URL" print if it fails
-        do_it <- force
+    if (!missing(savepoint)) {
+        warning("savepoint has been deprecated. This option no longer does anything.")
+    }
+    if (!missing(autorollback)) {
+        warning("autorollback has been deprecated. This option no longer does anything.")
     }
-    do_it(crPOST(batches_url, body=toJSON(body)))
+
```
Comment (nealrichardson) [inline R/batches.R]: Retain the comment for why `suppressMessages` is here.

File: R/append-dataset.R (hunk 2 above)
Comment (jonkeane) [reply R/append-dataset.R]: This is done, but GitHub's magic for hiding comments that are addressed was foiled by it being a line below.

**Comment by jonkeane** [comment]
I've updated with these changes, so this is ready for another round.

**Pull request merged**